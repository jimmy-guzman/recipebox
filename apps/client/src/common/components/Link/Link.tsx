import * as React from 'react'
import { css } from '@emotion/react'

import { COLOR_TEXT_BASE, COLOR_TEXT_ALT } from '../constants'

export interface LinkProps {
  children: React.ReactNode
  target?: string
  to: string
}

export const linkCss = css`
  text-decoration: none;
  color: ${COLOR_TEXT_BASE};
  &:hover {
    color: ${COLOR_TEXT_ALT};
    cursor: pointer;
  }
`

export const Link = ({ children, target, to }: LinkProps): JSX.Element => {
  return (
    <a href={to} target={target} rel='noreferrer' css={linkCss}>
      {children}
    </a>
  )
}
