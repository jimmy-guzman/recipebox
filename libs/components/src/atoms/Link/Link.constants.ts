import { css } from '@emotion/react';

import { COLOR_TEXT_ALT, COLOR_TEXT_BASE } from '../../constants';

export const linkCss = css`
  text-decoration: none;
  color: ${COLOR_TEXT_BASE};
  &:hover {
    color: ${COLOR_TEXT_ALT};
    cursor: pointer;
  }
`;
