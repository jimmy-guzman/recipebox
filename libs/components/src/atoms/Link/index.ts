export { Link, type LinkProps } from './Link';
export { linkCss } from './Link.constants';
