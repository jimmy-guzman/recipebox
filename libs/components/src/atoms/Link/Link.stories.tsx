import { Story, Meta } from '@storybook/react'

import { Link, LinkProps } from './Link'

export default {
  title: 'components/atoms/Link',
  component: Link,
} as Meta

const Template: Story<LinkProps> = (args) => (
  <Link {...args}>I am an Link!</Link>
)

export const Primary = Template.bind({})
