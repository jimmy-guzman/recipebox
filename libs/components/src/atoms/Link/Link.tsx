import * as React from 'react';

import { linkCss } from './Link.constants';

export interface LinkProps {
  children: React.ReactNode;
  target?: string;
  to: string;
}

export const Link = ({ children, target, to }: LinkProps): JSX.Element => {
  return (
    <a href={to} target={target} rel='noreferrer' css={linkCss}>
      {children}
    </a>
  );
};
