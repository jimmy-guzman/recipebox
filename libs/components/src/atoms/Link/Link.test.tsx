import { render } from '@testing-library/react'

import { Link } from './Link'

describe('<Link />', () => {
  it('should render', () => {
    const { container } = render(<Link to='#'>I am a link!</Link>)

    expect(container.firstChild).toMatchSnapshot()
  })
})
