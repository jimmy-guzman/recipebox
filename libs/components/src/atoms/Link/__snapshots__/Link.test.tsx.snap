// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> should render 1`] = `
.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #bf268a;
}

.emotion-0:hover {
  color: #04d9b2;
  cursor: pointer;
}

<a
  class="emotion-0"
  href="#"
  rel="noreferrer"
>
  I am a link!
</a>
`;
